core/parentchain/block-import-dispatcher/src/triggered_dispatcher.rs

```diff
@@ -182,10 +183,9 @@ where
 		let blocks_to_import =
 			self.import_queue.pop_until(predicate).map_err(Error::ImportQueue)?;
 
-		let events_to_import = self
-			.events_queue
-			.pop_from_front_until(blocks_to_import.len())
-			.map_err(Error::ImportQueue)?;
+		let depth = min(self.events_queue.peek_queue_size().unwrap_or(0), blocks_to_import.len());
+		let events_to_import =
+			self.events_queue.pop_from_front_until(depth).map_err(Error::ImportQueue)?;
```

Comment:
not sure if this is legit. It does fix the "Cannot Pop..." error. but may just hide an underlying issue?

Reply:
I think this is the underlying issue maybe. Does your error only happen during initial sync?

Reply:
no, the error happens when the first block should be properly imported after initial sync

Reply:
Hmm, there are still a few blocks between initial sync and and the creation checkpoint usually, but I don't know why this issue only exists on the first block.

Reply:
So, I do think there is an underlying issue, but this hack should not break anything.